clemenscodes/njvm
# finish a2

- remove init method and print init message on instantiation
- update integration tests for v2
- fix njvm integration test
- adjust help and version output
- reading file from command line works
- decode instructions from bin file
- print the instructions correctly from bin file
- a2/prog1 and prog2.out added to tests dir
- add integration tests for a2, currently failing
- dynamically grow stack with vec, adjusted unit tests
- adjust opcode unit tests to new instructions
- refactor codebase
- move njvm to seperate file
- format
- check version number of binary, convert to functional style
- use global variable for version number
- now working on implementing the new instructions to make tests pass
- use dependency injection to test stdin
- use dependency injection to test wrint and stdout
- njvm itself generic over io, not just some methods
- some refactoring before a bigger refactoring
- some refactoring
- generic stack and program memory
- refactoring, errorhandling
- decompose read file function to make it easier to unit test
- add integration tests for programs from a1
- refactoring, now testing
- export utils in external lib
- remove util lib because tests wouldnt run, add tests for utils
- add utils back
- implement more unit tests
- add static data area
- implement default trait for ninjavm and processor
- refactored, now tdd
- format
- implement pushg and popg
- improved util unit testing
- better read_file unit test
- some more error handling
- implement asf, rsf, pushl, popl
- removed expect statements and handle errors with exit instead
- unit test asf and rsf
- finish unit testing, added some more asm files, now more integration testing
- add integration test
- add integration test
- finish a2
- bump version to 0.2.0
